src/node_buffer.cc

```diff
@@ -701,6 +931,53 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void IsNull(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!HasInstance(args[0]))
+    return env->ThrowTypeError("first arg should be a Buffer");
+
+  CHECK(args[0]->IsObject());
+  ARGS_THIS(args[0].As<Object>())
+
+  uint32_t offset = args[1]->Uint32Value();
+  CHECK_LE(offset, obj_length);
+
+  char* ptr = obj_data + offset;
+
+  args.GetReturnValue().Set(ptr == nullptr);
```

Comment:
That wasn't what I meant but I like that even better.

My suggestion was more along the lines of either persisting the nullptr buffer and then checking that `Local<Value>::New(isolate, null_singleton)->StrictEquals(args[0])` or creating the nullptr buffer from a `static char null_singleton[]` and checking that `Buffer::Data(args[0]) == null_singleton`.